test/order-c.js

```diff
@@ -0,0 +1,4 @@
+import { log } from "./order-tracker";
+import "./order-a";
+log("c");
+import "./order-b";
```

Comment:
I see, you're expecting this `import "./order-b"` declaration to be hoisted above the `log("c")` call. I think that's a problem that would be solved by hoisting `import` declarations, or simply restricting `import` declarations so that they must appear before other statements/declarations, so this ordering would be forbidden by the compiler.